packages/foam-core/src/model/workspace.ts

```diff
@@ -129,14 +129,17 @@ export class FoamWorkspace implements IDisposable {
     note: Note,
     link: NoteLink
   ): URI {
-    let targetUri: URI | null = null;
+    let targetUri: URI | undefined;
```

Comment:
- Is this union on `undefined` really necessary? Couldn't we just add an empty URI?
- If it is necessary, shouldn't line 157 be updated to deal with it?

Reply:
typescript is smart enough to see that by line 157 the type has become just URI. I like that, it better describes what is actually happening